Incomplete Tests

I did just a `git pull` on `master` in order to _attempt_ to publish a new version of the package
and when I try to run `npm version` I see the following error message:
```
npm ERR! ERROR: Coverage for branches (94.12%) does not meet global threshold (100%)
```
![image](https://user-images.githubusercontent.com/194400/73274535-9e70ec80-41dd-11ea-937a-24e5e015284d.png)

We need to write at _least_ one new test for the new functionality added by #70 for #69 
to ensure that any _future_ refactoring/updates do not create regressions.

Running `npm test` we see that the "uncovered" lines are _branches_ on `118` and `121`:
![image](https://user-images.githubusercontent.com/194400/73274768-07f0fb00-41de-11ea-8ee6-933a01b284cf.png)

Viewing the coverage report in the browser confirms my suspicion:
hapi-error/coverage/lcov-report/index.js.html
![image](https://user-images.githubusercontent.com/194400/73276097-0d4f4500-41e0-11ea-95bb-f28b852ee13c.png)
The uncovered lines relate to the `redirect` if statements (_branches_) not having an `else` ...

Also, it appears that Codecov is not tracking our test coverage ... 
no coverage comment in the PR #70 and no recent tracking on the repo! 🤷‍♂ 
https://codecov.io/github/dwyl/hapi-error?branch=master
![image](https://user-images.githubusercontent.com/194400/73275001-6c13bf00-41de-11ea-8c12-83dcd3385d9f.png)

This is because our `.travis.yml` is not doing it's job ... 
https://github.com/dwyl/hapi-error/blob/c33c335dcd3a0fcab83c21d6679d756e53afb946/.travis.yml#L1-L6

the `after_success` command should be updated to:
```yml
after_success:
  - bash <(curl -s https://codecov.io/bash)
```

Let's fix this ...

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,6 +1,7 @@
 language: node_js
 node_js:
   - "8"
+  - "node"
 sudo: false
 after_success:
-	npm run codecov
+  - bash <(curl -s https://codecov.io/bash)

--- a/lib/index.js
+++ b/lib/index.js
@@ -112,20 +112,18 @@ exports.plugin = {
           // custom redirect https://github.com/dwyl/hapi-error/issues/5
           var currentCodeConfig = config.statusCodes[statusCode];
           if (currentCodeConfig && currentCodeConfig.redirect) {
-            // if parameter is function, process it and redirect exactly as given (unless false or null)
+            // if redirect is function invoke it with the request object
             if (isFunction(currentCodeConfig.redirect)) {
-              let redirectPath = currentCodeConfig.redirect(request);
-              if (redirectPath) {
-                return reply.redirect(redirectPath);
-              }
-            } else if (currentCodeConfig.redirect) { // if parameter is string, append redirect query
+              return reply.redirect(currentCodeConfig.redirect(request));
+            } 
+            else { 
+              // if parameter is string, append redirect query
               var redirectString = request.url.pathname + request.url.search;
               return reply.redirect(currentCodeConfig.redirect + '?redirect=' + redirectString);
             }
           }
 
-          if (currentCodeConfig
-            && currentCodeConfig.message) {
+          if (currentCodeConfig && currentCodeConfig.message) {
             msg = isFunction(currentCodeConfig.message)
               ? currentCodeConfig.message(msg, request)
               : currentCodeConfig.message

--- a/package.json
+++ b/package.json
@@ -9,7 +9,6 @@
     "dev": "PORT=8000 ./node_modules/.bin/nodemon example/server_example.js",
     "start": "node example/server_example.js",
     "check-coverage": "npm run test && nyc check-coverage --statements 100 --functions 100 --lines 100 --branches 100",
-    "codecov": "nyc report --reporter=text-lcov > coverage.lcov tape ./test/*.test.js && codecov",
     "lint": "node_modules/.bin/goodparts ./lib"
   },
   "repository": {
@@ -26,7 +25,6 @@
   },
   "devDependencies": {
     "boom": "^7.2.0",
-    "codecov": "^3.0.4",
     "decache": "^4.1.0",
     "good": "^8.1.1",
     "good-console": "^7.1.0",
@@ -69,7 +67,15 @@
   "nyc": {
     "exclude": [
       "example/*.js",
+      "coverage/*",
       "test/*.js"
-    ]
+    ],
+    "report-dir": "./coverage",
+    "reporter": [
+      "lcov",
+      "text"
+    ],
+    "cache": false,
+    "all": true
   }
 }

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "hapi-error",
-  "version": "2.2.0",
+  "version": "2.2.1",
   "description": "catch errors in your hapi application and display the appropriate error message/page",
   "main": "lib/index.js",
   "scripts": {

